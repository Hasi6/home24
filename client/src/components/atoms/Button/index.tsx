import styled from 'styled-components';

export const Button = styled.button`
  padding: 10px;
  border: none;
  background-color: ${({ theme }) => theme.colors.platinum600};
  color: white;
  cursor: pointer;
  border-radius: 6px;

  &:hover {
    background-color: ${({ theme }) => theme.colors.platinum800};
  }
`;
