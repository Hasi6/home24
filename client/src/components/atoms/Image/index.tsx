import styled from 'styled-components';

export const Image = styled.img`
  width: 100%;
  height: 150px;
  object-fit: cover;

  @media ${(props) => props.theme.media.sm} {
    height: 200px;
  }

  @media ${(props) => props.theme.media.md} {
    height: 250px;
  }
`;
