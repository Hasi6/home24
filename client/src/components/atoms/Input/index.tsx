import React, { InputHTMLAttributes } from 'react';
import styled from 'styled-components';

type Props = InputHTMLAttributes<HTMLInputElement>;

const StyledInput = styled.input`
  padding: 8px 15px;
  border: 2px solid #ccc;
  border-radius: 5px;
  font-size: 16px;
  color: #333;
  background-color: #fff;

  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);

  ::placeholder {
    color: #aaa;
  }

  &:focus {
    outline: none;
    border-color: #0070f3;
    box-shadow: 0 2px 5px rgba(0, 112, 243, 0.2);
  }

  &:disabled {
    background-color: #f5f5f5;
    color: #bbb;
  }
`;

export function Input(props: Props) {
  return <StyledInput type='text' {...props} />;
}
