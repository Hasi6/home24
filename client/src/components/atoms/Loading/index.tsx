import styled from 'styled-components';

export const Loading = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  font-size: 24px;
  color: #666;
`;
