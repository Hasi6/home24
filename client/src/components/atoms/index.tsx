import { Button } from '@/components/atoms/Button';
import { Image } from '@/components/atoms/Image';
import { Text } from '@/components/atoms/Text';
import { Loading } from '@/components/atoms/Loading';
import { Input } from '@/components/atoms/Input';

export { Button, Image, Text, Loading, Input };
