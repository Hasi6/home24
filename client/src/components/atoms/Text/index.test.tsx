import React from 'react';
import 'jest-styled-components';

import { Text } from '@/components/atoms';
import { render } from '@/utils/test/render';

describe('Text component', () => {
  it('renders correctly', () => {
    const { container } = render(<Text />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('has correct margin styling', () => {
    const { getByText } = render(<Text>Sample Text</Text>);
    const textComponent = getByText('Sample Text');
    expect(textComponent).toHaveStyleRule('margin', '5px 0');
  });
});
