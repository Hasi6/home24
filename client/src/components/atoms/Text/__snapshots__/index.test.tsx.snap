// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component renders correctly 1`] = `
.c0 {
  width: 100%;
  height: 150px;
  object-fit: cover;
}

@media (min-width: 768px) {
  .c0 {
    height: 200px;
  }
}

@media (min-width: 992px) {
  .c0 {
    height: 250px;
  }
}

<img
  class="c0"
/>
`;

exports[`Text component renders correctly 1`] = `
.c0 {
  margin: 5px 0;
}

<p
  class="c0"
/>
`;
